docs/migrating.rst

```diff
@@ -0,0 +1,235 @@
+
+    from google.cloud import ndb
+
+    client = ndb.Client()
+
+    with context as client.context():
+        do_something_with_ndb()
+
+Note that the example above is assumming the google credentials are set in
+the environment.
+
+Keys
+====
+
+There are some methods from the ``key`` module that are not implemented in
+this version of `ndb`:
+
+    - Key.from_old_key.
+    - Key.to_old_key.
+
+Models
+======
+
+There are some methods from the ``model`` module that are not implemented in
+this version of `ndb`. This is because getting the indexes relied on GAE
+context functionality:
+
+    - get_indexes.
+    - get_indexes_async.
+
+Properties
+==========
+
+There are various small changes in some of the model properties that might
+trip you up when migrating code. Here are some of them, for quick reference:
+
+- The `BlobProperty` constructor only sets `_compressed` if explicitly
+  passed. The original set `_compressed` always.
+- In the exact same fashion the `JsonProperty` constructor only sets
+  `_json_type` if explicitly passed.]
+- Similarly, the `DateTimeProperty` constructor only sets `_auto_now` and
+  `_auto_now_add` if explicitly passed.
+- `TextProperty(indexed=True)` and `StringProperty(indexed=False)` are no
+  longer supported.
+- The `Property()` constructor (and subclasses) originally accepted both
+  `unicode` and `str` (the Python 2 versions) for `name` (and `kind`) but now
+  only accept `str`.
+
+QueryOptions and Query Order
+============================
+
+The QueryOptions class from ``google.cloud.ndb.query``, has been reimplemented,
+since ``google.appengine.datastore.datastore_rpc.Configuration`` is no longer
+available. It still uses the same signature, but does not support original
+Configuration methods.
+
+Similarly,b ecause ``google.appengine.datastore.datastore_query.Order`` is no
+longer available, the ``ndb.query.PropertyOrder`` class has been created to
+replace it.
+
+MessageProperty and EnumProperty
+================================
+
+These properties, from the ``ndb.msgprop`` module, depend on the Google
+Protocol RPC Library, or `protorpc`, which is not an `ndb` dependency. For
+this reason, they are not part of this version of `ndb`.
+
+Tasklets
+========
+
+When writing a `tasklet`, it is no longer necessary to raise a Return
+exception for returning the result. A normal return can be used instead::
+
+    @ndb.tasklet
+    def get_cart():
+        cart = yield CartItem.query().fetch_async()
+        return cart
+
+Note that "raise Return(cart)" can still be used, but it's not recommended.
+
+There are some methods from the ``tasklet`` module that are not implemented in
+this version of `ndb`, mainly because of changes in how an `ndb` context is
+created and used in this version:
+
+    - add_flow_exception.
+    - make_context.
+    - make_default_context.
+    - QueueFuture.
+    - ReducedFuture.
+    - SerialQueueFuture.
+    - set_context.
+    - toplevel.
```

Comment:
I think we can still implement ``toplevel``. Or did you look at it and find out we couldn't?

Reply:
I only took a quick glance while going through the TODO list, and I thought its reliance on _make_cloud_datastore_context meant it was out. I didn't stop to think about the implementation. Do you think I should remove it from this list?

Reply:
I think it's doable. Let's keep this on our TODO list, still.